fabric-item-api-v1/src/testmod/java/net/fabricmc/fabric/test/item/CustomDamageTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.item;
+
+import java.util.function.Consumer;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.PickaxeItem;
+import net.minecraft.item.ToolMaterials;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.item.v1.CustomDamageHandler;
+import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
+
+public class CustomDamageTest implements ModInitializer {
+	@Override
+	public void onInitialize() {
+		Registry.register(Registry.ITEM, new Identifier("fabric-item-api-v1-testmod", "weird_pickaxe"), new WeirdPick());
+	}
+
+	public static final CustomDamageHandler WEIRD_DAMAGE_HANDLER = new CustomDamageHandler() {
```

Comment:
This is related to the comment above, but like I said this kind of implementation for the custom damage handler just seems clunky, just checking if we're able to avoid it or not with lambdas.

Ideally I'd love to just be able to do something like
```java
super(
    ToolMaterials.GOLD,
    1,
    -2.8F,
    new FabricItemSettings()
        .customDamage((stack, amount, entity, callback) -> {
            if (entity.isSneaking()) {
                return amount;
            } else {
                CompoundTag tag = stack.getOrCreateTag();
                tag.putInt("weird", tag.getInt("weird") + 1);
                return 0;
            }
        })
);
```